tasks/sep/fp-recovery/005-set-game-implementation/justfile

```diff
@@ -0,0 +1,23 @@
+set positional-arguments
+
+# default recipe to display help information
+default:
+  @just --list
+
+# Generate the `input.json` with the game type and implementation to set.
+generate-input gameType='' implAddr='':
+  #!/usr/bin/env bash
+  set -euo pipefail
+  . .env
+  SET_GAME_TYPE_SIG="setImplementation(uint32, address)"
+  ENCODED_CALL=$(cast calldata "${SET_GAME_TYPE_SIG}" "{{gameType}}" "{{implAddr}}")
+
+  REGISTRY_DIR="../../../../lib/superchain-registry"
```

Comment:
Should we just read the latest registry values directly with curl instead of requiring a local checkout? This way there's no risk the user has an outdated local version

Reply:
We'd need to know where to find the SystemConfig or similar starting point to read the values from on-chain though and that would still come from the superchain-registry.  It may still be better though since the superchain-registry version here may not have been updated since they deployed fault proofs so wouldn't have the DisputeGameFactory address but would have SystemConfig that we could get the DGF address from.

Reply:
Yea perhaps we just make an HTTP request with curl or similar to fetch the system config address, then we can fetch the DGF and other addresses we need from that one

Reply:
I was able to load the SystemConfig address via a curl to the superchain-registry and the DisputeGameFactory from there.

I could also load the `OWNER_SAFE` from the superchain-registry `ProxyAdminOwner` field. From there I can call `getOwners()` get the address of the foundation and council safes, but I don't know which is which. The `NestedSignFromJson.sol` file can be updated to not need them, but the parent `nested.just` needs the env vars set for foundation and council safes so it knows which to use.

I've left it with `NestedSignFromJson.sol` just generically handling all the nested safe owners so it doesn't need to know the specific foundation and council addresses, and the `OWNER_SAFE` env var is populated from the superchain-registry.  We still need the `COUNCIL_SAFE` and `FOUNDATION_SAFE` env vars to make `nested.just` happy though. Not sure if we should just keep `NestedSignFromJson.sol` simpler until `nested.just` doesn't need the council and foundation safe addresses - the new code might be overkill (and it allows foundation safe owners to have no code which wasn't previously allowed).

Reply:
> but I don't know which is which

The only way I can think of to infer this is with a check based the threshold and/or number of owners: The foundation is a 5 of 7, and the SC is a 10 of 13. 

`nested.just` may always need some way to know which safe is foundation and which is the council, because signers need to specify which safe they are signing for. Though now that I think of it, in theory we could take the signer address as input, determine which safe has that signer (assuming no duplicate signers across safes), and figure out which safe they need to sign with

I think what you have is ok, and perhaps we just add an assertion that `ownerSafe` has 2 (different) signers, and that each signer has code

Reply:
Interesting, I think this is bit overkill to try to remove the:  
`COUNCIL_SAFE`
`FOUNDATION_SAFE`
Even if this unlikely that the 1 signer is owner on both safe (for privacy) as the security council member are public. 
This will leak the identity of the safe owner. 
However, I do think keeping them here for the signers is with clear name like this are explicit and good to keep imo.